Support for ECDSAP256SHA256 and ECDSAP384SHA384

AndreasDilworth opened the pull request:
This PR contains support for ECDSAP256SHA256 and ECDSAP384SHA384 among a few other critical fixes that still hasn't made it into a release.

Commit:
Fix trailer line formatting for debian changelogs in update-version.sh

Commit:
Update compatibility levels for debhelper

Commit:
Add support for ECDSA256 and ECDSA384 algorithms for DNSSEC

Commit:
Fix for #14792 atomiapowerdnssync with 16.04 with latest PDNS from repo

Commit:
Fix for #17429 Question regarding PowerDNS agent

mladenst commented on powerdns_sync/lib/Atomia/DNS/PowerDNSDatabase.pm:
```diff
@@ -150,7 +150,7 @@ sub add_zone {
 			$query = "INSERT INTO domains (name, type) VALUES ($name, '$zone_type')";
 			$self->dbi->do($query) || die "error inserting domain row: $DBI::errstr";
-			$domain_id = $self->dbi->last_insert_id(undef, undef, "domains", undef) || die "error retrieving last_insert_id";
+			$domain_id = $self->dbi->selectrow_array("SELECT LAST_INSERT_ID() FROM domains");
```
@AndreasDilworth why this instead of dbi->last_insert_id?

AndreasDilworth replied:
There is a support ticket number in the commit message. This was the fix we got proposed from Aleksandar Skodric way back when that ticket was open. I would presume that last_insert_id() does return undef otherwise and I'm not sure why!
Grepping through the file I see one more instance of last_insert_id usage, but I don't think we're hitting this case in our code.
An option could perhaps be to use the mysql_insertid property?

AndreasDilworth replied:
I have now traced this issue down properly. Cause was that we're issuing pings every time DBI is used. I have also tested that things are still working fine when database connection goes away without all those pings.

mladenst replied:
I can't find sufficient information in #14792. Was there a communication outside of this ticket that I could gain access to (because I don't see the proposed fix form Aleksandar Skodric)?

Ping does seem to wipe the last insert id from the client according to:
https://dev.mysql.com/doc/refman/5.7/en/mysql-insert-id.html

However, I don't feel comfortable removing the ping altogether without detailed testing. Could you provide more details on how you performed the test of the server going away?

AndreasDilworth replied:
Aleksandar sent that patch (which contained some extra print lines and stuff which I did remove) to Kjetil Jensen in private noting that he'd share the fix right away so we could get our systems operational while waiting for the fix to be released in the official package.

My testing wasn't extremely extensive. I terminated the SQL server while running powerdnssync and verified that outstanding transactions where rolled back and connection reestablished after the SQL server came back up again.
Log lines will look like this:
```
Jan 29 11:59:16 dns atomiapowerdnssync[31534]: DBD::mysql::db selectall_arrayref failed: MySQL server has gone away at /usr/share/perl5/Atomia/DNS/PowerDNSDatabase.pm line 313.
Jan 29 11:59:16 dns atomiapowerdnssync[31534]: Issuing rollback() due to DESTROY without explicit disconnect() of DBD::mysql::db handle database=powerdns;host=localhost;port=3306 at /usr/bin/atomiapowerdnssync line 233.
Jan 29 11:59:16 dns atomiapowerdnssync[31534]: Caught exception handling command: DBD::mysql::db selectall_arrayref failed: MySQL server has gone away at /usr/share/perl5/Atomia/DNS/PowerDNSDatabase.pm line 313.
```

So in conclusion: the two options we have are removing the pings (or finding another condition in which they should be sent) or using the method Aleksandar had for retrieving insert ID's.
Aleksandar's method is what we've been using in production for over a year.

mladenst replied:
@mlgli suggested the following change - make ping exclusion optional (by an optional parameter in sub dbi). Then using this optional ping exclusion when calling last_insert_id.

@mlgli can you amend this PR with such a change?

mlgli replied:
Yes, I can do it tomorrow.

mlgli replied:
Hi @AndreasDilworth , I made a pull request towards your fork, when it's merged there, the commit will appear here as well.
https://github.com/OnlineGroupEurope/atomiadns/pull/1

AndreasDilworth replied:
This looks like a good compromise to me!

Commit:
Revert "Fix for #14792 atomiapowerdnssync with 16.04 with latest PDNS from repo"

This reverts commit 5704f6e80037a4c89cacb76831d72fa25f38e17c.

Commit:
Fix performance and last_insert_id by not issuing ping on every use of DBI.

mlgli committed:
PDNS: Add an option to skip DB ping

In mysql, pinging a database causes last_insert_id not to work.
This changes so that ping is skipped when using dbi with last_insert_id

Commit:
Merge pull request #1 from mlgli/develop

PDNS: Add an option to skip DB ping (fixes last_insert_id)
